src/bindings/python/flux/asyncio/events.py

```diff
@@ -0,0 +1,66 @@
+###############################################################
+# Copyright 2022 Lawrence Livermore National Security, LLC
+# (c.f. AUTHORS, NOTICE.LLNS, COPYING)
+#
+# This file is part of the Flux resource manager framework.
+# For details, see https://github.com/flux-framework.
+#
+# SPDX-License-Identifier: LGPL-3.0
+###############################################################
+
+import asyncio
+
+import flux
+import flux.core.watchers
+
+from .selector import FluxSelector
+
+# Make sure submit async and we see job go through
+# Ensure the job and loop use the same handle
+handle = flux.Flux()  # pylint: disable=invalid-name
+
+
+async def submit(fluxjob):
+    """
+    Submit an async Flux Job.
+
+    Example usage
+    =============
+    # ensure you've started flux
+    import asyncio
+    from flux.asyncio import loop
+    import flux.job
+
+    fluxsleep = flux.job.JobspecV1.from_command(['sleep', '2'])
+    fluxecho = flux.job.JobspecV1.from_command(['echo', 'pancakes'])
+
+    tasks = [
+       loop.create_task(flux.asyncio.submit(fluxecho)),
+       loop.create_task(flux.asyncio.submit(fluxsleep)),
+    ]
+
+    asyncio.set_event_loop(loop)
+    results = loop.run_until_complete(asyncio.gather(*tasks))
+    # [JobID(456004999315456), JobID(456004999315457)]
+    """
+    uid = await flux.job.submit_async(handle, fluxjob)
+    return uid
+
+
+class FluxEventLoop(asyncio.SelectorEventLoop):
+    """
+    An asyncio loop that handles running Flux.
+    """
+
+    def __init__(self, flux_handle=None):
+        if not flux_handle:
+            flux_handle = handle
```

Comment:
I originally did that so submit() and the loop would have the same handle - is that not important? E.g., flux.Flux() wherever you call it / multiple times returns the same handle?

Reply:
No, you would get a different handle every time. I guess I was thinking---if they strictly need to be the same handle, then why is `flux_handle` accepted as an optional argument?

Reply:
Update for here - it stops working when you don't use a common handle for the submit and the loop.

Reply:
That's an "advanced" use case - technically someone could create some custom handle and then provide to their submit and loop. My expectation (why I say advanced) is that most people won't. But I think it could be an edge case, and we can easily support it with this method.

Reply:
If we want multiple handles to work, we'll need to re-design select a bit, and... well, everything else.  There needs to be a handle for the Selector instance that it will always add its watchers to.  If we want to allow flux futures to be incorporated that use different handles, we'll need to detect that case when they get added and never actually block in reactor_run but loop over them instead.  Unless, @grondo or @jameshcorbett do we have a way to add a handle to another reactor?

Reply:
Yes, in the C API we have [`flux_handle_watcher_create(3)`](https://flux-framework.readthedocs.io/projects/flux-core/en/latest/man3/flux_handle_watcher_create.html)

Reply:
Excellent, then yeah we could actually support that reasonably, we just have to figure out how to make sure that we get that done when a future wrapped in a task gets added to the event loop. Maybe we add a check into the __await__ method to see if the loop uses the same handle and if not add the future’s handle to it across the yield?

Reply:
okay so conceptually - the event loop has one internal handle, and when a new task is added, when we hit the future.__await__ we say:

1. is this the same flux_handle the loop has? If yes, carry on!
2. if no, we need to switch the handle to be the loop's (e.g., future.flux_handle = loop.handle) OR do we need to somehow steal the watchers for that handle and add to the one loop handle, or something else?

Reply:
okay - so I played with this for a bit, and came up with https://github.com/researchapps/flux-core/compare/test/asyncio...researchapps:flux-core:test/asyncio-updates?expand=1. The issue that we run into is that even though we can check for handle equality and then create the new HandleWatcher, there is still no connection back to the file descriptors and SelectKey that asyncio has registered, so the resulting outcome is that the loop runs forever because it can never know that it's ready. I chat with @trws and we decided that trying to get this working would be going down a rabbit hole of potentially huge complexity, and for now we should add a check to ensure that the handles are the same, and throw an error if not. And we can come back to this in a round 2 if we absolutely find a case where two different flux handles must be supplied.